scripts/compatibility_scanner/compatibility_scanner.q

```diff
@@ -0,0 +1,239 @@
+DEBUG_SHOW_REGEX_PASSED:0b;
+
+ZS_TO_CHECK:(  // .z functions to look out for without the ".z." prefix attached
+  "pm";
+  "zd";
+  "ac";
+  "bm";
+  "exit";
+  "pc";
+  "pd";
+  "pg";
+  "ph";
+  "pi";
+  "po";
+  "pp";
+  "pq";
+  "ps";
+  "pw";
+  "ts";
+  "vs";
+  "wc";
+  "wo";
+  "ws"
+ );
+
+getFullPath:{[path]
+  if[0~count path;:()];
+  if[0h~type path;
+    $[all 10h=type each path;:{x where {not x~()}each x}getFullPath each path;
+      [-2"ERROR: Wrong type for 'path': ",-3!type path;'wrongtype]];
+  ];
+  if[not 10h~type path;-2"ERROR: Wrong type for 'path': ",-3!type path;'wrongtype];
+
+  :@[{first system"readlink -f '",x,"'"};path;
+    {[x;y] -2"ERROR: Could not find path to: '",x,"'";exit 1}[path]];
+ };
+
+MAIN_SCRIPT_DIR:{("/" sv -1 _ "/" vs x),"/"}getFullPath string .z.f;  // Used this so that the script will load its dependencies correctly even if the user starts the script form another directory
+
+ASSIGNMENT_CHECKS_TSV:MAIN_SCRIPT_DIR,"assignment_checks.tsv";
+COMMANDS_CHECKS_TSV:MAIN_SCRIPT_DIR,"commands_checks.tsv";
+
+
+run:{[]
+  args:parseArgs[];
+  checksDict:loadChecksDict[];
+
+  args[`file]:$[
+    args`regex;getFilesFromRegex[args`file;args`dir];
+    ignoreNonQScripts getFullPath args`file
+  ];
+  args[`exclude]:$[
+    args`regex;getFilesFromRegex[args`exclude;args`dir];
+    ignoreNonQScripts getFullPath args`exclude
+  ];
+  if[10h~type args`dir;  // Checking directory exists if dir argument was passed, if not exits with an error
+    isInvalidDir:(()~args`dir)or @[{11h<>type key hsym`$x};args`dir;1b];
+    if[isInvalidDir;-2"ERROR: Directory '",args[`dir],"' could not be found";exit 1];
+  ];
+
+  if[not ()~args`dir;args[`file]:distinct args[`file],getDirFileList args`dir];
+  if[0<count args`file;args[`file]:filterExcluded[args`file;args`exclude]];
+  
+  lines:$[
+    0<>count args`file;[
+      -1"Scanning ",string[count args`file]," .q script(s) . . .\n";
+      raze scanFile[;checksDict] each args[`file]
+    ];
+    [-1"No files to scan";()]
+  ];
+
+  -1"\nChecked ",string[count args`file]," .q script(s)";
+  -1"Total lines with possible incompatibilities: ",string count lines;
+
+  $[
+    (not ()~args[`csv])and 0<>count lines;writeToCsv[args`csv;checksDict;lines];
+    0~count lines;-1"WARN: Skipped writing to csv, no lines to output";
+    ()
+  ];
+
+  exit 0;
+ };
+
+getFilesFromRegex:{[regexList;dir]
+  if[0~count regexList;:()];
+
+  if[dir~();dir:first system"pwd"];
+  res:raze{system"find ",x," -regextype posix-extended -type f -regex '",y,"'"}[dir]each regexList;
+
+  :getFullPath ignoreNonQScripts res;
+ };
+
+ignoreNonQScripts:{[files]
+  res:files where files like "*[.]q";
+  if[count[res]<>count files;-1"WARN: Ignored files that are not .q scripts"];
+
+  :res;
+ };
+
+scanFile:{[file;checksDict]
+  res:system"printf '%s\\n' \"$(grep -nHE '",combineRegexList[value checksDict],"' '",file,"')\"";
+  
+  if[""~raze/[res];res:()];
+  if[0<count res;
+    -1"\n" sv res;
```

Comment:
Will this print every line found to screen? Might be better to have a progress tracker like with your categorising. Output is being saved to a csv, don't see a need to print it here.

Reply:
Should I make it still print them out if no csv argument is passed or should the csv argument be required do you think?

Reply:
Resolved this with [8e2d972](https://github.com/DataIntellectTech/TorQ-Amazon-FinSpace-Starter-Pack/pull/81/commits/8e2d972521cdd880c4928fcb72c8e49a0834e22a)